doc/tutorials/02-charged_system/02-charged_system-1.ipynb

```diff
@@ -380,68 +680,129 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Plot the distribution functions\n",
-    "plt.figure(figsize=(10, 6), dpi=80)\n",
-    "plt.plot(r[:], rdf_00[:], label='Na$-$Na')\n",
-    "plt.plot(r[:], rdf_01[:], label='Na$-$Cl')\n",
-    "plt.xlabel('$r$', fontsize=20)\n",
-    "plt.ylabel('$g(r)$', fontsize=20)\n",
-    "plt.legend(fontsize=20)\n",
-    "plt.show()"
+    "def add_salt(system, anion_params, cation_params):        \n",
+    "    for _ in range(anion_params['number']):\n",
+    "        system.part.add(pos=np.random.random(3) * system.box_l, \n",
+    "                        type=anion_params['type'], q=-anion_params['valency'])\n",
+    "        \n",
+    "    for _ in range(cation_params['number']):\n",
+    "        system.part.add(pos=np.random.random(3) * system.box_l, \n",
+    "                        type=cation_params['type'], q=cation_params['valency'])\n",
+    "    "
    ]
   },
   {
-   "cell_type": "markdown",
+   "cell_type": "code",
+   "execution_count": null,
    "metadata": {},
+   "outputs": [],
    "source": [
-    "## 6 Task - Real Units\n",
+    "anion_params = {'type':3,\n",
+    "                'valency':2,\n",
+    "                'number':500}\n",
+    "cation_params = {'type':4,\n",
+    "                 'valency':2,\n",
+    "                 'number':500}\n",
+    "rod_length = 21\n",
+    "#TODO\n",
+    "steps_per_frame = 100\n",
+    "N_frames = 500\n",
+    "\n",
+    "total_anion_charge = -anion_params['number']*anion_params['valency']\n",
```

Comment:
If the slice is kept on particle creation, their charge can be queried directly rather than be assumed
`np.sum(an_ions.q)`